flink-ml-servable-lib/src/main/java/org/apache/flink/ml/classification/logisticregression/LogisticRegressionModelServable.java

```diff
@@ -0,0 +1,148 @@
+        List<Double> predictionResults = new ArrayList<>();
+        List<DenseVector> rawPredictionResults = new ArrayList<>();
+
+        int featuresColIndex = input.getIndex(getFeaturesCol());
+        for (Row row : input.collect()) {
+            Vector features = (Vector) row.get(featuresColIndex);
+            Tuple2<Double, DenseVector> dataPoint = predictOneDataPoint(features, coefficient);
+            predictionResults.add(dataPoint.f0);
+            rawPredictionResults.add(dataPoint.f1);
+        }
+
+        input.addColumn(getPredictionCol(), DataTypes.DOUBLE, predictionResults);
+        input.addColumn(
+                getRawPredictionCol(), DataTypes.VECTOR(BasicType.DOUBLE), rawPredictionResults);
+
+        return input;
+    }
+
+    public LogisticRegressionModelServable setModelData(InputStream... modelDataInputs)
+            throws IOException {
+        Preconditions.checkArgument(modelDataInputs.length == 1);
+
+        DataInputViewStreamWrapper inputViewStreamWrapper =
+                new DataInputViewStreamWrapper(modelDataInputs[0]);
+
+        DenseVectorSerializer serializer = new DenseVectorSerializer();
+        coefficient = serializer.deserialize(inputViewStreamWrapper);
+
+        return this;
+    }
+
+    public static LogisticRegressionModelServable load(String path) throws IOException {
+        LogisticRegressionModelServable servable =
+                ServableReadWriteUtils.loadServableParam(
+                        path, LogisticRegressionModelServable.class);
+
+        try (InputStream fsDataInputStream = ServableReadWriteUtils.loadModelData(path)) {
+            DataInputViewStreamWrapper dataInputViewStreamWrapper =
+                    new DataInputViewStreamWrapper(fsDataInputStream);
+            DenseVectorSerializer serializer = new DenseVectorSerializer();
+            servable.coefficient = serializer.deserialize(dataInputViewStreamWrapper);
+            return servable;
+        }
+    }
+
+    /**
+     * The main logic that predicts one input data point.
+     *
+     * @param feature The input feature.
+     * @param coefficient The model parameters.
+     * @return The prediction label and the raw probabilities.
+     */
+    public static Tuple2<Double, DenseVector> predictOneDataPoint(
+            Vector feature, DenseVector coefficient) {
+        double dotValue = BLAS.dot(feature, coefficient);
+        double prob = 1 - 1.0 / (1.0 + Math.exp(dotValue));
+        return Tuple2.of(dotValue >= 0 ? 1. : 0., Vectors.dense(1 - prob, prob));
+    }
+
+    /**
+     * Serializes the model data into byte array which can be saved to external storage and then be
+     * used to update the Servable by `TransformerServable::setModelData` method.
+     *
+     * @param modelData The model data to be serialized.
+     * @return The serialized model data in byte array.
+     */
+    public static byte[] serialize(DenseVector modelData) throws IOException {
```

Comment:
Does this mean that users need to know the internal data structure/type of the model data in order to use this public method?